pkg/reconciler/knativeserving/common/resources.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	mf "github.com/manifestival/manifestival"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes/scheme"
+	servingv1alpha1 "knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
+)
+
+// ResourceRequirementsTransform configures the resource requests for
+// all containers within all deployments in the manifest
+func ResourceRequirementsTransform(instance *servingv1alpha1.KnativeServing, log *zap.SugaredLogger) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() == "Deployment" {
+			deployment := &appsv1.Deployment{}
+			if err := scheme.Scheme.Convert(u, deployment, nil); err != nil {
+				return err
+			}
+			containers := deployment.Spec.Template.Spec.Containers
+			for i := range containers {
+				containers[i].Resources = instance.Spec.Resources[containers[i].Name]
```

Comment:
This will not only drop limits if empty, but also `requests`, which are important for scheduling decisions. I'm wondering if we want to only override those if explicitly set and keep the default requests in the manifest.

Reply:
Easy enough. Won't change defaults, but does expose knobs for tweaking.

Reply:
+1, I would expect a field by field merge here so that empty values in the operator CR don't trash field that are set in the manifest but not the operator CR

Reply:
Done